### DISCUSSION TITLE
Fine-tuning PaddleOCR with Custom Numbers-Only Dictionary

### DISCUSSION THREAD

**Discussion by Buligon** [PaddlePaddle/PaddleOCR#15178] [Q&A] [answered]
I'm working on a project where I only need to recognize numerical digits (0–9). I was wondering if it's possible to fine-tune an existing PaddleOCR model using a custom dictionary that includes only numbers, in order to reduce the number of possible output classes and improve accuracy/speed.

- Is it possible to restrict or retrain the model with a numbers-only character dictionary?
- If so, what would be the recommended way to approach this (e.g., modifying dict.txt, changing model configs, retraining with custom data)?
- Are there any pre-trained models or best practices for such a use case?

Any insights or references would be greatly appreciated!

**Comment by ManikSinghSarmaal** [comment, marked as answer]
Hey,
Yes it is possible to restrict  or retrain an existing model with a numbers-only character dictionary. You just have to train the model on your custom dict for your custom data, the recommended approach would be to confine your dict in the config file to the characters you want to recognize and finetune the model (PP-OCRv3 or PP-OCRv4) on your custom data.
This doc would help you - [Finetune.md](https://github.com/PaddlePaddle/PaddleOCR/blob/3cc4ae9f37c92459a2951be004194a807c9a781d/doc/doc_en/finetune_en.md#L4)
Also you can get pre-trained models form [Model List](https://github.com/Mushroomcat9998/PaddleOCR/blob/main/doc/doc_en/models_list_en.md#english-recognition-model) - This model supports both English and number recognition. Considering your case you can take a english pretrained model and then finetune on numerical digits(0-9).

**Reply by Buligon** [reply]
Thanks for your response!

Do you happen to know any repository or good tutorial with a practical example of this fine-tuning process? Having a reference like that would be really helpful.

Also, after fine-tuning the model, would it be ready for use with the standard inference arguments? For example, just by updating rec_char_dict_path to point to the new dictionary and rec_model_dir to the folder containing the training output?

Thanks in advance!
